membership/network.lua

```diff
@@ -1,75 +1,79 @@
-#!/usr/bin/env tarantool
-
 local ffi = require('ffi')
 local bit = require('bit')
 
-ffi.cdef([[
-struct ifaddrs {
-    struct ifaddrs  *ifa_next;    /* Next item in list */
-    char            *ifa_name;    /* Name of interface */
-    unsigned int     ifa_flags;   /* Flags from SIOCGIFFLAGS */
-    struct sockaddr *ifa_addr;    /* Address of interface */
-    struct sockaddr *ifa_netmask; /* Netmask of interface */
-    union {
-        struct sockaddr *ifu_broadaddr;  /* Broadcast address of interface */
-        struct sockaddr *ifu_dstaddr;    /* Point-to-point destination address */
-    } ifa_ifu;
-    void            *ifa_data;    /* Address-specific data */
-};
-
-struct in_addr {
-    uint32_t s_addr;
-};
-
-enum {
-    IFF_UP          = 0x1,  /* interface is up              */
-    IFF_BROADCAST   = 0x2,  /* broadcast address valid      */
-    IFF_POINTOPOINT = 0x10  /* interface is has p-p link    */
-};
-
-enum {
-    AF_INET         = 2     /* Internet IP Protocol         */
-};
-
-const char *strerror(int errno);
-int getifaddrs(struct ifaddrs **ifap);
-void freeifaddrs(struct ifaddrs *ifa);
-const char *inet_ntop(int af, const void *src,
-                      char *dst, socklen_t size);
-]])
-
-if ffi.os == "Linux" then
+local stash = require('membership.stash')
+if not stash.get('network.cdef_getifaddrs') then
     ffi.cdef([[
```

Comment:
Just pcall this part of code as it was done here - https://github.com/tarantool/avro-schema/commit/bbcf5fc14e3fd1962351574d3f6c30e84bcf7c92 / https://github.com/tarantool/avro-schema/commit/5e1ed3db0d69ab84ff0fb998f3fc9c5845cd33fe

Reply:
I don't like this approach because it doesn't show errors at all. I've tried to implement something like `if not pcall(ffi.typeof) then ffi.cdef()`, but it didn't work for `struct sockaddr`. Because it's always true even on bare tarantool, I don't know why.

Reply:
ok